test/e2e/pkg/infra/infra.go

```diff
@@ -0,0 +1,84 @@
+package infra
+
+import (
+	"context"
+
+	e2e "github.com/tendermint/tendermint/test/e2e/pkg"
+)
+
+// TestnetInfra provides an API for manipulating the infrastructure of a
+// specific testnet.
+type TestnetInfra interface {
```

Comment:
As mentioned in https://github.com/tendermint/tendermint/pull/8754#discussion_r897313891, in this PR all I've done is extract the Docker-specific functionality and hide it behind an interface. It's already the bare minimum interface surface area with the current architecture. For example, you'll notice there's no `Start` function on the `TestnetInfra` interface - that's because the `Start` function in the `main` package of the runner just uses `StartNode` in a loop.

Reply:
Ok your call. I just ideally want to have the interface be as simple as possible